axelor-production/src/main/java/com/axelor/apps/production/service/ManufOrderWorkflowService.java

```diff
@@ -94,6 +95,9 @@ public ManufOrder plan(ManufOrder manufOrder) throws AxelorException {
     }
 
     manufOrder.setPlannedEndDateT(this.computePlannedEndDateT(manufOrder));
+    if(!ObjectUtils.isEmpty(manufOrder.getBillOfMaterial()) && !ObjectUtils.isEmpty(manufOrder.getBillOfMaterial().getUnit())) {
```

Comment:
no needs to check isEmpty , just check != null and only one time (means don't check for unit).